.github/workflows/tests.yml

```
@@ -0,0 +1,54 @@
+name: Tests
+
+on:
+  pull_request:
+    branches:
+      - "*"
+  push:
+    branches:
+      - master
+
+jobs:
+  ruby_test:
+    name: Ruby Test Action
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby-version: [2.6.6, 2.7.2, "3.0"]
+
+    services:
+      postgres:
+        image: postgres:12
+        ports: ["5432:5432"]
+        env:
+          POSTGRES_PASSWORD: password
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+      - name: Bundle install
+        run: |
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+      - name: Install postgres
+        run: |
+          sudo apt-get -yqq install libpq-dev
+      - name: Create Dummy  and Migrate Database
```

Just to be a bit more precise:

```suggestion
      - name: Create db and run tests
```

Once we trim down the dependencies we won't need a db 🙂